src/Microsoft.NetCore.Analyzers/Core/Security/UseXmlReaderBase.cs

```diff
@@ -15,6 +15,16 @@ namespace Microsoft.NetCore.Analyzers.Security
 {
     public abstract class UseXmlReaderBase : DiagnosticAnalyzer
     {
+        private static readonly LocalizableString s_Message = new LocalizableResourceString(
```

Comment:
@LingxiaChen @dotpaul This PR by itself seems correct to me, but seems like there is a bug in the original implementation. Why are all the DiagnosticDescriptors in sub-types of `UseXmlReaderBase` defined with the same message and description. Do we want all these diagnostics to have the same message, but different ID and title? Seems weird to me. Is this just a copy paste mistake in original implementation?

Reply:
Ah, the [message string](https://github.com/dotnet/roslyn-analyzers/blob/89ca12075bfa80be5e3872e88f955060835fe61e/src/Microsoft.NetCore.Analyzers/Core/Security/SystemSecurityCryptographyResources.resx#L301) seems to be formatted to mention different unsafe overloads and the [description](https://github.com/dotnet/roslyn-analyzers/blob/89ca12075bfa80be5e3872e88f955060835fe61e/src/Microsoft.NetCore.Analyzers/Core/Security/SystemSecurityCryptographyResources.resx#L283) seems to be a generic message applicable to all overloads. Makes sense to me.

Reply:
Also wanted different help link URLs for each scenario, so users don't have to dig through documentation for the relevant case.